### DISCUSSION TITLE
404 Error

### DISCUSSION THREAD

**Discussion by DmitriyMalayev** [vercel/community#7180] [Help] [answered]
### Summary

Trying to deploy my app, all steps successful but site has 404 error. Any help appreciated

![image](https://github.com/vercel/community/assets/56614284/48448a70-41ae-4458-9ed5-87d7253427a8)

### Example

crud-app-ashy-delta.vercel.app

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/56614284/48448a70-41ae-4458-9ed5-87d7253427a8)

**Comment by versecafe** [comment]
@DmitriyMalayev What route are you deploying since it seems you only have one, all that shows is that `/` is a 404, if for example you only deployed to `/api` then this is perfectly normal and is working as expected.

**Reply by DmitriyMalayev** [reply]
I just made it public. 
The routes are in server/routes
https://github.com/DmitriyMalayev/crud-app/

**Reply by DmitriyMalayev** [reply]
not sure if this is correct

{
    "version": 2,
    "builds": [{"src": "app.js", "use": "@vercel/node" }],
    "routes": [{"src": "server/routes/(.*)", "dest": "/app.js"}]
}

**Reply by DmitriyMalayev** [reply]
@versecafe any tips?

**Comment by amyegan** [comment, marked as answer]
Hey @DmitriyMalayev. The `builds` and `routes` you're using are legacy properties and not recommended for most projects. We have a guide walking through [how to deploy an Express.js app with Vercel](https://vercel.com/guides/using-express-with-vercel). I recommend reading through those steps and modifying your repo to match the recommended configuration.

I've also shared some general tips for debugging 404 errors over in https://github.com/vercel/community/discussions/6980#discussioncomment-9705033

[a comment by an automated account is not shown]
